### ISSUE TITLE
pip install -e not appearing in pip list

### ISSUE THREAD

**Issue by fangzhouli** [googlecolab/colabtools#2539]
**Describe the current behavior**
``pip install -e .`` resulting in ModuleNotFoundError.

I tried ``pip list``, noticing that editables are not listed. If I remove -e flag, it will reappear in the pip list. If relevant, I am calling ``! python`` to run the CLI script. (It had been working until 2 weeks ago.)

**What web browser you are using**
Chrome

**Additional context**
Link to a minimal, public, self-contained notebook that reproduces this issue.
 - Share the file using your GitHub account using  File > Save a copy as a GitHub Gist.
 - or Share Drive notebooks using the Share button then 'Get Shareable Link'.
https://colab.research.google.com/drive/1I3KBFS1z8QKN5Drs3ji9V6oM6st0Mdz2?usp=sharing

**Comment by craigcitro** [comment]
Thanks for the report!

There is definitely a bug here: modules installed with `!pip install -e` aren't visible: the `.egg-link` file being created is not on `sys.path`. (Naively adding it also doesn't do the trick.) The initial `import food_ke` statement is working because it's being executed from the directory containing the `food_ke` folder.

Is there a reason you need to install with `-e`?

**Comment by fangzhouli** [comment]
Hi, thanks for the quick response!

Yes, .egg-link is in site-packages, but not on sys.path. Naively adding it to sys.path allowing IPython importation, but seems not working for !python CLI call.

The main reason will be that development will be much easier, and we prefer the data to be accessible locally rather than site-packages.

Is this a recent bug? I believe I haven't encountered this issue before Xmas.

**Comment by craigcitro** [comment]
We didn't purposely change anything related to this, and there were no pip/setuptools updates that I see, so I'm not sure why you're newly seeing it.

If you tweak `os.environ['PYTHONPATH']` does the `!python` call work?

**Comment by fangzhouli** [comment]
I just changed os.environ['PYTHONPATH'] to a random string, and !python still worked.

So the Python I'm calling is not using the default path for some reason?

**Comment by craigcitro** [comment]
That's not how `PYTHONPATH` works: garbage entries there get added to otherwise-sensible ones.

**Comment by fangzhouli** [comment]
seems like if I changed the ``PYTHONPATH`` to ``/root/FoodAtlas`` where we put the ``git cloned`` repo in, the ModuleNotFoundError was resolved.

How was it connected to ``pip install -e`` not visible?

**Comment by craigcitro** [comment]
Tracking internally as b/213499612, though we're not actively working on it.

@fangzhouli let us know if you're still blocked.

**Comment by fangzhouli** [comment]
Yes, just tried it again, install via -e is not put into the list for my pkg.

One observation is that when ``pip install`` usually logs ``Successfully installed {pkg name}-{version}``, but in this case, the result only shows ``Successfully installed {pkg name}``. Seems like metadata in setup.py is ignored for some reason?

No rush, but I'm curious to know what is going on. Thanks for helping out Craig!

**Comment by robinnarsinghranabhat** [comment]
Almost a year. Any updates on this ? Facing the same problem

**Comment by Pchatain** [comment]
Also running into this problem. Any updates?
